src/jdk.compiler/share/classes/com/sun/tools/javac/main/JavaCompiler.java

```diff
@@ -1637,7 +1640,7 @@ public void generate(Queue<Pair<Env<AttrContext>, JCClassDecl>> queue, Queue<Jav
                 }
                 if (results != null && file != null)
                     results.add(file);
-            } catch (IOException ex) {
+            } catch (IOException | FileSystemNotFoundException | InvalidPathException | ReadOnlyFileSystemException ex) {
```

Comment:
If we're hypothesizing plausible unchecked exceptions, maybe consider adding UncheckedIOException ?

I'd also recommend one exception per line, breaking the line before the `|`

```
} catch (IOException 
    | FileSystemNotFoundException 
    | InvalidPathException 
    | ReadOnlyFileSystemException 
    | UncheckedIOException ex) {
```